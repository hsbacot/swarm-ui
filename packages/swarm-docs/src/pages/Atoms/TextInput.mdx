import { TextInput } from '@meetup/swarm-components';
import Example from '../../components/examples/textInput';
import PropsTable from '../../components/propstable';

# TextInput
---
A wrapper for an html `<input>` element with an optional `<label>` tag.

<br/>
<br/>

### Snippet
---

Use this interactive snippet to try the component:
<br/>

```.jsx
<TextInput id="input" value="phife dawg two" />
```

<br/>
<br/>

### Props
---
<br/>
<PropsTable propMetaData={TextInput.__docgenInfo} />

<br/>
<br/>

### a11y
---
<br/>

Keyboard interactions: **n/a**

<br/>
Must supply a <b>label</b> html tag with an <b>`htmlFor`</b> React attribute that equals the value of the textInput's <b>id</b>.

<br/>
<br/>

### Examples
---
<br/>

<Example />
