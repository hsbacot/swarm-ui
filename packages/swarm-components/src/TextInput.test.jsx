import React from 'react';
import { shallow } from 'enzyme';
import { TextInput } from './TextInput';

describe('TextInput', () => {
	const mockOnChange = jest.fn();

	const testCases = [
		['Default (empty)', <TextInput key="1" id="default" name="default" value="" />],
		[
			'Default (value)',
			<TextInput key="2" id="default" name="default" value="foo" />,
		],
		[
			'Disabled',
			<TextInput key="3" id="disabled" name="disabled" value="" disabled />,
		],
		['Error', <TextInput key="4" id="error" name="error" value="" error />],
		[
			'isSearch',
			<TextInput key="5" id="isSearch" name="isSearch" value="" isSearch />,
		],
		[
			'Pattern',
			<TextInput
				key="6"
				id="pattern"
				name="pattern"
				value=""
				pattern="[A-Za-z]{3}"
			/>,
		],
		[
			'onChange',
			<TextInput
				key="7"
				id="onChange"
				name="onChange"
				value=""
				onChange={mockOnChange}
			/>,
		],
		[
			'Icon',
			<TextInput key="8" id="icon" name="icon" value="" iconShape="location-pin" />,
		],
		[
			'Icon (custom size)',
			<TextInput
				key="9"
				id="iconsize"
				name="iconsize"
				value=""
				iconShape="location-pin"
				iconSize="m"
			/>,
		],
		[
			'maxLength (char count)',
			<TextInput
				key="10"
				id="maxLength"
				name="maxLength"
				value="foo fooo"
				maxLength={20}
			/>,
		],
	];

	test.each(testCases)('Snapshot: %s', (description, element) => {
		expect(shallow(element)).toMatchSnapshot();
	});

	describe('maxLength', () => {
		it('should render error state when maxLength exceeded', () => {
			const textInput = shallow(<TextInput name="foo" value="" maxLength={3} />);
			textInput.setProps({ value: 'new value' });
			expect(textInput.exists('[data-swarm-text-input="error"]')).toEqual(true);
		});
	});
});
