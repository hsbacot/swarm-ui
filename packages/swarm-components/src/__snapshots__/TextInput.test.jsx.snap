// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Snapshot: Default (empty) 1`] = `
<div
  data-swarm-text-input-wrapper="default"
>
  <input
    data-swarm-text-input="default"
    id="default"
    name="default"
    type="text"
    value=""
  />
</div>
`;

exports[`TextInput Snapshot: Default (value) 1`] = `
<div
  data-swarm-text-input-wrapper="default"
>
  <input
    data-swarm-text-input="default"
    id="default"
    name="default"
    type="text"
    value="foo"
  />
</div>
`;

exports[`TextInput Snapshot: Disabled 1`] = `
<div
  data-swarm-text-input-wrapper="default"
>
  <input
    data-swarm-text-input="disabled"
    disabled={true}
    id="disabled"
    name="disabled"
    type="text"
    value=""
  />
</div>
`;

exports[`TextInput Snapshot: Error 1`] = `
<div
  data-swarm-text-input-wrapper="default"
>
  <input
    data-swarm-text-input="error"
    id="error"
    name="error"
    type="text"
    value=""
  />
</div>
`;

exports[`TextInput Snapshot: Icon (custom size) 1`] = `
<div
  data-swarm-text-input-wrapper="icon"
>
  <input
    data-swarm-text-input="default"
    iconSize="m"
    id="iconsize"
    name="iconsize"
    type="text"
    value=""
  />
  <span
    data-swarm-input-icon="location-pin"
  >
    <Icon
      aria-hidden={true}
      shape="location-pin"
    />
  </span>
</div>
`;

exports[`TextInput Snapshot: Icon 1`] = `
<div
  data-swarm-text-input-wrapper="icon"
>
  <input
    data-swarm-text-input="default"
    id="icon"
    name="icon"
    type="text"
    value=""
  />
  <span
    data-swarm-input-icon="location-pin"
  >
    <Icon
      aria-hidden={true}
      shape="location-pin"
    />
  </span>
</div>
`;

exports[`TextInput Snapshot: Pattern 1`] = `
<div
  data-swarm-text-input-wrapper="default"
>
  <input
    data-swarm-text-input="default"
    id="pattern"
    name="pattern"
    pattern="[A-Za-z]{3}"
    type="text"
    value=""
  />
</div>
`;

exports[`TextInput Snapshot: isSearch 1`] = `
<div
  data-swarm-text-input-wrapper="default"
>
  <input
    data-swarm-text-input="default"
    id="isSearch"
    name="isSearch"
    type="search"
    value=""
  />
</div>
`;

exports[`TextInput Snapshot: maxLength (char count) 1`] = `
<div
  data-swarm-text-input-wrapper="default"
>
  <input
    data-swarm-text-input="default"
    id="maxLength"
    name="maxLength"
    type="text"
    value="foo fooo"
  />
  <CharCount
    charLength={8}
    maxLength={20}
  />
</div>
`;

exports[`TextInput Snapshot: onChange 1`] = `
<div
  data-swarm-text-input-wrapper="default"
>
  <input
    data-swarm-text-input="default"
    id="onChange"
    name="onChange"
    onChange={[MockFunction]}
    type="text"
    value=""
  />
</div>
`;
